cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ArtifactHttpHandlerInternal.java

```diff
@@ -128,12 +131,15 @@ public void getArtifactBytes(HttpRequest request, HttpResponder responder,
       .add(HttpHeaderNames.LAST_MODIFIED, newModifiedDate.format(DateTimeFormatter.RFC_1123_DATE_TIME));
 
     String lastModified = request.headers().get(HttpHeaderNames.IF_MODIFIED_SINCE);
+    //We truncate milliseconds from timestamps when comparing modified them.
```

Comment:
"when comparing modified them" -> "when comparing modified time"

Also maybe elaborate on why this is needed